cdci_data_analysis/analysis/queries.py

```diff
@@ -523,25 +523,19 @@ def get_query_products(self,instrument,job,run_asynch,query_type='Real',logger=N
             logger.error("passing request issue: %s", e)
             raise
 
-        except Exception as e: # TODO: could we avoid these? they make error tracking hard
+        except Exception as e:
+
+            # TODO: could we avoid these? they make error tracking hard
             # TODO we could use the very same approach used when test_communication fails
             logger.exception("failed to get query products")
 
             #status=1
             job.set_failed()
-
             if os.environ.get('DISPATCHER_DEBUG', 'yes') == 'yes':
                 raise
+            e_message = getattr(e, 'message', None)
 
-            e_message = getattr(e, 'message', '')
-            debug_message = repr(e) + ' : ' + getattr(e, 'debug_message', '')
-
-            query_out.set_failed('get_dataserver_products found job failed',
-                                 logger=logger,
-                                 sentry_client=sentry_client,
-                                 excep=e,
-                                 e_message=e_message,
-                                 debug_message=debug_message)
+            raise InternalError(e_message)
```

Comment:
yeah, but class also matters! What if it catches `ValueError` ? Will be quite hard to debug!